wormhole-connect/src/routes/ntt/nttBase.ts

```diff
@@ -0,0 +1,317 @@
+import { BigNumber } from 'ethers';
+import {
+  ChainId,
+  ChainName,
+  TokenId,
+} from '@wormhole-foundation/wormhole-connect-sdk';
+import { Route, TokenConfig } from 'config/types';
+import {
+  SignedMessage,
+  UnsignedMessage,
+  UnsignedNTTMessage,
+  SignedNTTMessage,
+  isSignedNTTMessage,
+} from '../types';
+import { fetchVaa } from 'utils/vaa';
+import { hexlify, parseUnits } from 'ethers/lib/utils.js';
+import { BaseRoute } from '../bridge';
+import { isEvmChain, toChainName, wh } from 'utils/sdk';
+import { TOKENS } from 'config';
+import { getTokenById, getTokenDecimals, removeDust } from 'utils';
+import { getNativeVersionOfToken } from 'store/transferInput';
+import { getPlatform } from './platforms';
+import { InboundQueuedTransfer } from './types';
+import { DestContractIsPausedError } from './errors';
+
+export abstract class NTTBase extends BaseRoute {
+  isSupportedChain(chain: ChainName): boolean {
+    return isEvmChain(chain) || chain === 'solana';
+  }
+
+  async isSupportedSourceToken(
+    token?: TokenConfig,
+    destToken?: TokenConfig,
+    sourceChain?: ChainName | ChainId,
+    destChain?: ChainName | ChainId,
+  ): Promise<boolean> {
+    if (!token || !sourceChain || !this.isSupportedToken(token, sourceChain)) {
+      return false;
+    }
+    if (
+      destChain &&
+      destToken &&
+      !this.isSupportedToken(destToken, destChain)
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  async isSupportedDestToken(
+    token?: TokenConfig,
+    sourceToken?: TokenConfig,
+    sourceChain?: ChainName | ChainId,
+    destChain?: ChainName | ChainId,
+  ): Promise<boolean> {
+    if (!token || !destChain || !this.isSupportedToken(token, destChain)) {
+      return false;
+    }
+    if (
+      sourceChain &&
+      sourceToken &&
+      !this.isSupportedToken(sourceToken, sourceChain)
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  isSupportedToken(token: TokenConfig, chain: ChainName | ChainId): boolean {
+    return (
+      this.isSupportedChain(token.nativeChain) &&
+      !!token.nttManagerAddress &&
+      toChainName(chain) === token.nativeChain
+    );
+  }
+
+  async isRouteSupported(
+    sourceToken: string,
+    destToken: string,
+    amount: string,
+    sourceChain: ChainName | ChainId,
+    destChain: ChainName | ChainId,
+  ): Promise<boolean> {
+    return Promise.all([
```

Comment:
maybe this could be more clear if we use await here

```suggestion
    try {
    
    const results = await Promise.all([
      this.isSupportedChain(wh.toChainName(sourceChain)),
      this.isSupportedChain(wh.toChainName(destChain)),
      this.isSupportedSourceToken(
        TOKENS[sourceToken],
        TOKENS[destToken],
        sourceChain,
        destChain,
      ),
      this.isSupportedDestToken(
        TOKENS[destToken],
        TOKENS[sourceToken],
        sourceChain,
        destChain,
      )]);
    return result.every(result => result);
    } catch (e: any) {
    // for debug porpouse
    console.trace(e);
    return false;
    }
```

Reply:
agreed